docs/source/notes/extending.rst

```diff
@@ -1,10 +1,238 @@
 Extending PyTorch
 =================
 
-In this note we'll cover ways of extending :mod:`torch.nn`,
+In this note we'll cover ways of extending :mod:`torch`, :mod:`torch.nn`,
 :mod:`torch.autograd`, and writing custom C extensions utilizing our C
 libraries.
 
+Extending :mod:`torch`
```

Comment:
I don't think this section should come first in the doc. nn is the most important extension point, autograd second. Totally new tensor types definitely is third on the list of likely things people want to do.